typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -26,21 +49,30 @@ async function main() {
 
   const sources = chains.filter((chain) => !skip || !skip.includes(chain));
   for (const source of sources) {
-    for (const destination of sources.slice().filter((d) => d !== source)) {
+    for (const destination of sources.filter((d) => d !== source)) {
+      const labels = {
+        origin: source,
+        remote: destination,
+        ...constMetricLabels,
+      };
       try {
         await sendMessage(app, source, destination);
+        messagesCount.labels({ ...labels, status: 'success' }).inc();
       } catch (err) {
         console.error(
           `Error sending message from ${source} to ${destination}, continuing...`,
           `${err}`.replaceAll('\n', ' ## '),
         );
         failureOccurred = true;
+        messagesCount.labels({ ...labels, status: 'failure' }).inc();
       }
       // Sleep 500ms to avoid race conditions where nonces are reused
       await sleep(500);
     }
   }
 
+  await submitMetrics();
```

Comment:
We could submit before sleeping, but then we are doing a lot more submissions since that is a loop and it is fairly traditional with push metrics to only report on job completion so this seems fine.